File: 0658-find-k-closest-elements/0658-find-k-closest-elements.py
class Solution(object):
    def findClosestElements(self, arr, k, x):
        """
        :type arr: List[int]
        :type k: int
        :type x: int
        :rtype: List[int]
        """
        left = 0
        right = len(arr) - k
        
        while left < right:
            mid = left + (right - left)/ 2
            if x - arr[mid] > arr[mid + k] - x:
                left = mid + 1
            else:
                right = mid
        
        ans = []
        
        for i in range(left, left + k):
            ans.append(arr[i])
        return ans

File: 0658-find-k-closest-elements/0658-find-k-closest-elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        List<Integer> ans = new ArrayList <Integer>();
        int left = 0;
        int right = arr.length - k;
        
        while (left < right){
            int mid = left + (right - left) / 2;
            if ( x - arr[mid] > arr[mid+k] -x){
                left = mid+1;
            }
            else{
                right = mid;
            }
        }
        arr = Arrays.copyOfRange(arr, left, left+k);
        for(int i = 0; i < arr.length; i++){
            ans.add(arr[i]);
        }
        return ans;
        
    }
}